overlord/snapstate/snapstate.go

```diff
@@ -1907,6 +1919,15 @@ func Remove(st *state.State, name string, revision snap.Revision, flags *RemoveF
 	// 'purge' flag disables automatic snapshot for given remove op
 	if flags == nil || !flags.Purge {
 		if tp, _ := snapst.Type(); tp == snap.TypeApp && removeAll {
+			if sz, err := EstimateSnapshotSize(st, name); err == nil {
+				if err := osutilCheckFreeSpace(dirs.SnapdVarDir(dirs.GlobalRootDir), uint64(sz)); err != nil {
```

Comment:
One can set `snapshot.automatic.retention=no` what prevents snapshots from being automatically created. In this case even if there's no enough space for the snapshot, removal should not fail. Since we fail here when there is not enough space, perhaps moving the size estimation check further down to the `else` branch after the call to `AutomaticSnapshot()` would be sufficient.

Reply:
Good catch, thanks!